specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-12-01/bms.json

```diff
@@ -5485,6 +5500,13 @@
           "$ref": "#/definitions/RetentionPolicy",
           "description": "Retention policy with the details on backup copy retention ranges."
         },
+        "tieringPolicy": {
+          "description": "Tiering policy to automatically move RPs to another tier\r\nKey is Target Tier, defined in RecoveryPointTierType enum.\r\nTiering policy specifies the criteria to move RP to the target tier.",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

Why is this modeled as additionalProperties? ARM doesn't allow additionalProperties unless the members are unknown, dynamic or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined)

Reply:
The tieringPolicy field is a dictionary. As per the OpenAPI specification, dictionaries need to be defined with the additionalProperties keyword.
Ref: https://swagger.io/docs/specification/data-models/dictionaries/

Reply:
Is the size of the dictionary dynamic? Meaning can the user only specify one element out of all the possible elements?

Reply:
Yes, user can specify a different TieringPolicy for each target tier